pkg/internal/apis/certmanager/v1alpha2/conversion.go

```diff
@@ -41,8 +41,24 @@ func Convert_v1alpha2_CertificateSpec_To_certmanager_CertificateSpec(in *v1alpha
 			out.PrivateKey = &certmanager.CertificatePrivateKey{}
 		}
 
-		out.PrivateKey.Algorithm = certmanager.PrivateKeyAlgorithm(in.KeyAlgorithm)
-		out.PrivateKey.Encoding = certmanager.PrivateKeyEncoding(in.KeyEncoding)
+		switch in.KeyAlgorithm {
+		case v1alpha2.ECDSAKeyAlgorithm:
+			out.PrivateKey.Algorithm = certmanager.ECDSAKeyAlgorithm
+		case v1alpha2.RSAKeyAlgorithm:
+			out.PrivateKey.Algorithm = certmanager.RSAKeyAlgorithm
+		default:
+			out.PrivateKey.Algorithm = certmanager.PrivateKeyAlgorithm(in.KeyAlgorithm)
```

Comment:
I wondered if this should be an error, if the KeyAlgorithm somehow got set to an unsupported value, but I suppose that'll be caught later by API validation and maybe by in-controller validation of the retrieved Certificate.

Reply:
Conversion cannot error - but yep, validation should handle this 😄

Reply:
I'll take your word for it, but note that the function signature does return an error.

Reply:
Ah, that is a fair point 😄 . In k/k, they cannot, but as this is a webhook, we actually can.

That said, returning an error is going to cause *big* issues for the cluster (i.e. stop garbage collection working thus blocking namespace deletion etc).

I think we should avoid returning errors *as much as possible*, and rely on validation to throw errors if/when needed 😄